cmd/internal/cli/delete.go

```diff
@@ -90,20 +94,53 @@ var deleteImageCmd = &cobra.Command{
 	Example: docs.DeleteExample,
 	Args:    cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		pullFrom := args[len(args)-1]
-		transport, ref := uri.Split(pullFrom)
-		if ref == "" {
-			sylog.Fatalf("Bad URI %s", pullFrom)
+		imageRef, err := library.NormalizeLibraryRef(args[0])
+		if err != nil {
+			sylog.Fatalf("Error parsing library ref: %v", err)
 		}
 
-		switch transport {
```

Comment:
@DrDaveD this switch can still be used to identify transports which do not yet support `delete` functionality

Reply:
I added a check to catch if someone tries to use an unsupported protocol scheme in a URI passed in as a delete imageRef.  Without that it was giving an incomprehensible error message.